src/libopenrave/libopenrave.cpp

```diff
@@ -791,7 +791,7 @@ class RaveGlobal : private boost::noncopyable, public boost::enable_shared_from_
         if( !_pdefaultsampler ) {
             boost::mutex::scoped_lock lock(_mutexinternal);
             BOOST_ASSERT( _mapenvironments.size() > 0 );
-            _pdefaultsampler = GetDatabase()->CreateSpaceSampler(_mapenvironments.begin()->second->shared_from_this(),"MT19937");
+            _pdefaultsampler = RaveCreateSpaceSampler(_mapenvironments.begin()->second->shared_from_this(), "MT19937");
```

Comment:
@undisputed-seraphim umm... why make this change?

Reply:
@undisputed-seraphim ping

Reply:
I removed the callee `CreateSpaceSampler`. Now I realize that `RaveCreateSpaceSampler` creates it under the global OpenRAVE instance which may not necessarily be correct.

I will do it like this:
```
InterfaceBasePtr ifBasePtr = GetDatabase()->Create(_mapenvironments.begin()->second->shared_from_this(), PT_SpaceSampler, "MT19937");
_pdefaultsampler = RaveInterfaceCast<SpaceSamplerBase>(ifBasePtr);
```